src/snyk/common/services/learnService.ts

```diff
@@ -0,0 +1,113 @@
+import axios from 'axios';
+import { isCodeIssue, isOssIssue, OpenCommandIssueType } from '../../common/commands/types';
+import type { completeFileSuggestionType } from '../../snykCode/interfaces';
+import { OssIssueCommandArg } from '../../snykOss/views/ossVulnerabilityTreeProvider';
+import { ILog } from '../logger/interfaces';
+
+const ecosystems = [
+  'all',
+  'java',
+  'javascript',
+  'python',
+  'golang',
+  'php',
+  'cpp',
+  'csharp',
+  'kubernetes',
+  'ruby',
+  'elixir',
+  'docker',
+  'npm',
+] as const;
+type LessonEcosystem = typeof ecosystems[number];
+
+const isValidEcosystem = (potentialEcosystem: string | null | undefined): potentialEcosystem is LessonEcosystem => {
+  return ecosystems.includes(potentialEcosystem as LessonEcosystem);
+};
+
+export type Lesson = {
+  title: string;
+  lessonId: string;
+  ecosystem: LessonEcosystem;
+  url: string;
+};
+
+type PackageMapper = {
+  [key in LessonEcosystem]?: string[];
+};
+
+export interface ILearnService {
```

Comment:
q: did we clarify with @kussberg  what we should return in case the lesson does not exist?

Reply:
Yup, had a chat with him about this on Friday:
```
The logic follows:
- If the issue is not a vulnerability (e.g. code quality issue). Don't show a link
- If Snyk Learn has a lesson matching the issue CWE, show a link to the lesson (ideally for the same ecosystem)
- If Snyk Learn doesn't have a lesson matching the issue CWE, show a link to Snyk Learn's homepage
```

Reply:
👍